test/FunctionalTests/Microsoft.OData.Core.Tests/NavigationPropertyOnComplexTests.cs

```diff
@@ -1652,5 +1652,27 @@ private static IEdmModel ContainmentComplexModel()
         }
 
         #endregion
+        [Fact]
+        public void NullValidationErrorMessageForCollectionsofComplexTypes()
+        {
+            string expectedErrorMessage = "A null value was found for the property named 'Addresses', which has the expected type 'Collection(DefaultNs.Address)[Nullable=False]'. The expected type 'Collection(DefaultNs.Address)[Nullable=False]' does not allow null values.";
+            string actualErrorMessage = "";
+            const string payload =
+                "{" +
+                    "\"@odata.context\":\"http://host/$metadata#People/$entity\"," +
+                    "\"UserName\":\"abc\"," +
+                    "\"Addresses\":null" +

```

Comment:
Nullable at collection is for the items in the collection. Collection itself is not nullalble. 
So, your test case is not correct.
It can be :  "Addresses": [ ]   // empty collection, but can't be null
or It can be: "Addresses": [ null  ]  // non empty collection with null item.

@mikepizzo need mike's input.

Reply:
As Sam says, nullable on a collection means it can *contain* a null value, not that it can *be* a null value. 

This test *correctly* validates that the collection value cannot be null, and the same should be true if Addresses is defined as a collection with nullable=true. I don't know if we have such a test, and it would probably be a breaking change to make that an error if it's not today, but that would be the expected behavior.

Reply:
@mikepizzo there was no such test. Have added the test though. Please review.

Reply:
:thumbs up:

---
In reply to: [439432201](https://github.com/OData/odata.net/pull/1793#discussion_r439432201) [](ancestors = 439432201)